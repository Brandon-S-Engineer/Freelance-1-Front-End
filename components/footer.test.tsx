import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from './footer';

describe('Footer Component', () => {
  it('renders the footer content correctly', () => {
    render(<Footer />);

    // Check for the presence of the copyright text
    expect(screen.getByText(/Â© 2024 eCommerce, Inc. All rights reserved/i)).toBeInTheDocument();
  });

  it('renders with the correct classes for styling', () => {
    render(<Footer />);

    // Check for specific class names
    const footerElement = screen.getByRole('contentinfo'); // Semantic role for a footer
    expect(footerElement).toHaveClass('bg-white');
    expect(footerElement).toHaveClass('border-t');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
