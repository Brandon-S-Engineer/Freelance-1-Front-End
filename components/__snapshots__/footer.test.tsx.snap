// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="bg-white border-t"
  >
    <div
      class="mx-auto py-10"
    >
      <p
        class="text-center text-sm text-black"
      >
        Â© 2024 eCommerce, Inc. All rights reserved
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
